pkg/operator/ceph/object/notification/obc_label_controller.go

```diff
@@ -0,0 +1,217 @@
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errors.Wrapf(err, "failed to retrieve ObjectBucketClaim %q", request.NamespacedName)
+	}
+
+	// find the namespace for the ceph cluster (may be different than the namespace of the topic CR)
+	// Make sure a CephCluster is present otherwise do nothing
+	cephCluster, isReadyToReconcile, cephClusterExists, reconcileResponse := opcontroller.IsReadyToReconcileNoName(
+		r.client,
+		r.context,
+		controllerName,
+	)
+	if !isReadyToReconcile {
+		// This handles the case where the Ceph Cluster is gone and we want to delete that CR
+		if !cephOBC.GetDeletionTimestamp().IsZero() && !cephClusterExists {
+			// Return and do not requeue. Successful deletion.
+			return reconcile.Result{}, nil
+		}
+		logger.Debug("Ceph cluster not yet present")
+		return reconcileResponse, nil
+	}
+	r.clusterSpec = &cephCluster.Spec
+
+	// DELETE: the CR was deleted
+	if !cephOBC.GetDeletionTimestamp().IsZero() {
+		logger.Debugf("ObjectBucketClaim %q was deleted", request.NamespacedName)
+		// Return and do not requeue. Successful deletion.
+		return reconcile.Result{}, nil
+	}
+
+	// Populate clusterInfo during each reconcile
+	r.clusterInfo, _, _, err = mon.LoadClusterInfo(r.context, cephCluster.Namespace)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to populate cluster info")
+	}
+
+	// create the notification if needed
+	return r.updateCephBucketNotifications(cephOBC)
+}
+
+func (r *ReconcileOBCLabels) updateCephBucketNotifications(cephOBC *bktv1alpha1.ObjectBucketClaim) (reconcile.Result, error) {
+	// step 1: get the ObjectBucket
+	namespace := cephOBC.Namespace
+	cephOB := bktv1alpha1.ObjectBucket{}
+	bucketName := types.NamespacedName{Namespace: namespace, Name: cephOBC.Spec.ObjectBucketName}
+	if err := r.client.Get(context.TODO(), bucketName, &cephOB); err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to retrieve ObjectBucket %q", bucketName)
+	}
+	objectStoreName, err := getCephObjectStore(cephOB)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to get object store from ObjectBucket %q", bucketName)
+	}
+
+	// step 2: delete all existing notifications
+	p := Provisioner{
+		Client:      r.client,
+		Context:     r.context,
+		ClusterInfo: r.clusterInfo,
+		ClusterSpec: r.clusterSpec,
+	}
+	sess, err := p.createSession(
+		cephOB.Spec.AdditionalState["cephUser"],
+		objectStoreName,
+	)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to create session for bucket notification provisioning for ObjectBucketClaim %q", bucketName)
+	}
+	err = p.DeleteAll(&cephOB, sess)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed delete all bucket notifications from ObjectbucketClaim %q", bucketName)
+	}
+
+	// Step 3: looking for notifications in the labels
```

Comment:
Steps may be added or removed in the future, so how about if we don't number them?